ee/vellum_ee/workflows/display/tests/workflow_serialization/test_manual_trigger_serialization.py

```diff
@@ -0,0 +1,113 @@
+        output = Inputs.input
+
+
+def create_workflow(trigger=None):
+    """Factory for creating test workflows."""
+
+    class TestWorkflow(BaseWorkflow[Inputs, BaseState]):
+        graph = trigger >> SimpleNode if trigger else SimpleNode
+
+        class Outputs(BaseWorkflow.Outputs):
+            output = SimpleNode.Outputs.output
+
+    return TestWorkflow
+
+
+def serialize(workflow_class) -> JsonObject:
+    """Helper to serialize a workflow."""
+    return get_workflow_display(workflow_class=workflow_class).serialize()
+
+
+def test_manual_trigger_serialization():
+    """Workflow with ManualTrigger serializes with trigger field."""
+    result = serialize(create_workflow(ManualTrigger))
+    workflow_raw_data = cast(JsonObject, result["workflow_raw_data"])
+    trigger = cast(JsonObject, workflow_raw_data["trigger"])
+
+    assert trigger["type"] == "MANUAL"
+    assert not DeepDiff(
+        {
+            "type": "MANUAL",
+            "definition": {"name": "ManualTrigger", "module": ["vellum", "workflows", "triggers", "manual"]},
+        },
+        trigger,
+        ignore_order=True,
+    )
+
+
+def test_no_trigger_serialization():
+    """Workflow without trigger has no trigger field."""
+    result = serialize(create_workflow())
+    workflow_raw_data = cast(JsonObject, result["workflow_raw_data"])
+    assert "trigger" not in workflow_raw_data
+
+
+def test_manual_trigger_multiple_entrypoints():
+    """ManualTrigger with multiple entrypoints."""
+
+    class NodeA(BaseNode):
+        class Outputs(BaseNode.Outputs):
+            output = Inputs.input
+
+    class NodeB(BaseNode):
+        class Outputs(BaseNode.Outputs):
+            output = Inputs.input
+
+    class MultiWorkflow(BaseWorkflow[Inputs, BaseState]):
+        graph = ManualTrigger >> {NodeA, NodeB}
+
+        class Outputs(BaseWorkflow.Outputs):
+            output_a = NodeA.Outputs.output
+            output_b = NodeB.Outputs.output
+
+    result = serialize(MultiWorkflow)
+    workflow_data = cast(JsonObject, result["workflow_raw_data"])
+    trigger = cast(JsonObject, workflow_data["trigger"])
+    nodes = cast(JsonArray, workflow_data["nodes"])
+
+    assert trigger["type"] == "MANUAL"
+    assert len([n for n in nodes if cast(JsonObject, n)["type"] == "GENERIC"]) >= 2
+
+
+def test_serialized_workflow_structure():
+    """Verify complete structure of serialized workflow."""
+    result = serialize(create_workflow(ManualTrigger))
+    workflow_raw_data = cast(JsonObject, result["workflow_raw_data"])
+    definition = cast(JsonObject, workflow_raw_data["definition"])
+
+    assert result.keys() == {"workflow_raw_data", "input_variables", "state_variables", "output_variables"}
+    assert workflow_raw_data.keys() == {"nodes", "edges", "display_data", "definition", "output_values", "trigger"}
+    assert definition["name"] == "TestWorkflow"
+
+
+def test_unknown_trigger_type():
+    """Unknown trigger types raise ValueError."""
```

Comment:
[oos] we will likely one day allow users to define their own triggers, so having some default serialization to support gets us a step closer there